transform-message-formats/main.bal

```diff
@@ -0,0 +1,90 @@
+import ballerina/http;
+import ballerina/log;
+
+configurable int port = 8290;
+configurable string hospitalServiceUrl = "http://localhost:9090";
+
+final http:Client hospitalServiceEP = check initializeHttpClient();
+
+function initializeHttpClient() returns http:Client|error => new (hospitalServiceUrl);
+
+type Patient record {|
+    string name;
+    string dob;
+    string ssn;
+    string address;
+    string phone;
+    string email;
+|};
+
+type RequestData record {|
+    *Patient;
+    string doctor;
+    string hospital_id;
+    string hospital;
+    string card_no;
+    string appointment_date;
+|};
+
+type ReservationRequest record {|
+    Patient patient;
+    string doctor;
+    string hospital;
+    string appointment_date;
+|};
+
+type Doctor record {|
+    string name;
+    string hospital;
+    string category;
+    string availability;
+    float fee;
+|};
+
+type ReservationResponse record {|
+    int appointmentNumber;
+    Doctor doctor;
+    Patient patient;
+    float fee;
+    string hospital;
+    boolean confirmed;
+    string appointmentDate;
+|};
+
+service /healthcare on new http:Listener(port) {
+    isolated resource function post categories/[string category]/reserve(RequestData payload)
+            returns ReservationResponse|http:NotFound|http:BadRequest|http:InternalServerError {
+        ReservationRequest req = transform(payload);
+        string hospitalId = payload.hospital_id;
```

Comment:
We are getting an error (https://github.com/ballerina-platform/ballerina-lang/issues/41293) and it breaks the LS if we do so, so I have selected this approach instead of inline access. I could use the destructuring, but it feels like adding an unnecessary code line.